### PR TITLE
feat: requirement building blocks

### PR DESCRIPTION
**Pull request by BrickheadJohnny**
Common requirement-related components & Storybook stories

### CODE REVIEW

File: src/components/requirements/DataBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+const DataBlock = ({
+  isLoading,
+  error,
+  children,
+}: PropsWithChildren<Props>): JSX.Element => {
+  if (isLoading) return <Skeleton className="inline w-24" />;
+
+  return (
+    <Tooltip open={error ? undefined : false}>
```
Comment (BrickheadJohnny) [inline src/components/requirements/DataBlock.tsx:23]: Actually, I'm not really sure if we need the tooltip & error state here, since we'll probably use it for static data only. What do you think?

File: src/components/requirements/ChainIndicator.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { Badge } from "@/components/ui/Badge";
+import { CHAINS } from "@/config/chains";
+import type { Register } from "wagmi";
+
+export const ChainIndicator = ({
+  chain,
+}: { chain: Register["config"]["chains"][number]["id"] }) => (
```
Comment (BrickheadJohnny) [inline src/components/requirements/ChainIndicator.tsx:7]: What do you think, should we extract the `Register["config"]["chains"][number]["id"]` to a type somewhere?

File: src/styles/globals.css [hunk 3]
```diff
@@ -63,6 +63,17 @@ html {
   --blue-800: #1e40af;
   --blue-900: #1e3a8a;
 
+  --orange-50: #fff7ed;
```
Comment (BrickheadJohnny) [inline src/styles/globals.css:66]: These are unrelated changes (we had a warning icon in the DataBlock component originally, I left it out this time), but we'll need the orange colors anyways, so I left these lines here.

File: src/components/requirements/ChainIndicator.tsx (hunk 2 above)
Comment (dominik-stumpf) [reply src/components/requirements/ChainIndicator.tsx:7]: It looks like a common type, we can extract it into a wagmi type file.

File: src/components/ui/Badge.tsx [hunk 4]
```diff
@@ -7,7 +7,7 @@ export const badgeVariants = cva(
   {
     variants: {
       size: {
-        sm: "text-xs h-5",
+        sm: "text-xs h-5 gap-1",
```
Comment (dominik-stumpf) [inline src/components/ui/Badge.tsx:10]: I think we could have this on all sizes like on button.

File: src/components/requirements/DataBlock.tsx
Comment (dominik-stumpf) [inline src/components/requirements/DataBlock.tsx:1]: Regarding nesting, I think this is as deep as we should go.

File: public/chainLogos/eth.svg [hunk 5]
```diff
@@ -0,0 +1 @@
...sianBlur in="shadowOffsetOuter1" result="shadowBlurOuter1" stdDeviation=".5"/><feComposite in="shadowBlurOuter1" in2="SourceAlpha" operator="out" result="shadowBlurOuter1"/><feColorMatrix in="shadowBlurOuter1" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.204257246 0"/></filter></defs><g fill="none" fill-rule="evenodd"><use fill="#000" filter="url(#eth-a)" xlink:href="#eth-b"/><use fill="#627EEA" xlink:href="#eth-b"/><use fill="url(#eth-c)" style="mix-blend-mode:soft-light" xlink:href="#eth-b"/><circle cx="16" cy="15" r="14.5" stroke="#000" stroke-opacity=".097"/><g fill-rule="nonzero"><use fill="#000" filter="url(#eth-d)" xlink:href="#eth-e"/><use fill="#FFF" fill-opacity="0" fill-rule="evenodd" xlink:href="#eth-e"/></g><g fill="#FFF" fill-rule="nonzero" transform="translate(9 3)"><polygon fill-opacity=".602" points="7.498 0 7.498 8.87 14.995 12.22"/><polygon points="7.498 0 0 12.22 7.498 8.87"/><polygon fill-opacity=".602" points="7.498 17.968 7.498 23.995 15 13.616"/><polygon points="7.498 23.995 7.498 17.967 0 13.616"/><polygon fill-opacity=".2" points="7.498 16.573 14.995 12.22 7.498 8.872"/><polygon fill-opacity=".602" points="0 12.22 7.498 16.573 7.498 8.872"/></g></g></svg>
```
[the remark this answers is not in the document]
Comment (BrickheadJohnny) [reply public/chainLogos/eth.svg:1]: Ooooh, this is a good suggestion, I'll take a look at it!
